### ISSUE TITLE
Add `groupByNes`

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats#2282]
Sometimes it'd be nice to group a Set to non-empty Sets, like:

```scala
Set("a", "aa", "aaa", "bbb").groupByNes(_.length) <-> SortedMap(
  1 -> NonEmptySet.one("a"),
  2 -> NonEmptySet.one("aa"),
  3 -> NonEmptySet.of("aaa", "bbb")
)
```

I suggest that a `SetSyntax` trait is added with syntax for this, similar to the existing `ListSyntax`.
The way `ListOps#groupByNel` is implemented (`list.toNel.groupBy(...)`)suggests that this could also involve having `groupBy` in `NonEmptySet`.

I'll gladly contribute that, given some pointers :)

**Comment by Obarros** [comment]
@kubukoz I can take a look on this one.

**Comment by kubukoz** [comment]
Sure, go for it :)

**Comment by kubukoz** [comment]
Closed in #2285
